tox.ini

```diff
@@ -0,0 +1,21 @@
+[tox]
+minversion = 3.11.0
+envlist =
+  ruff
+  py{311,312}
+isolated_build = true
+skip_missing_interpreters = true
+
+[testenv]
+deps = poetry
+skip_install = true
+sitepackages = false
+commands_pre =
+  poetry install --all-extras
+commands =
+  pytest -o 'addopts=--cov --cov-config .coveragerc --cov-report term --cov-report xml --cov-report html' tests/
+
+[testenv:ruff]
```

Comment:
I see you have ruff here. Looking at this, it seems to be a nice tool, does it also replace black?

Reply:
Ohh… forgot to add this. I think new version can take care of formatting, but I have to check.